projects/framework-nuke/extensions/plugins/nuke_node_collector.py

```diff
@@ -0,0 +1,35 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+import nuke
+
+from ftrack_framework_core.plugin import BasePlugin
+from ftrack_framework_core.exceptions.plugin import PluginExecutionError
+
+
+class NukeNodeCollectorPlugin(BasePlugin):
+    name = 'nuke_node_collector'
+
+    def ui_hook(self, payload):
+        '''
+        Return all available write nodes in Nuke
+        '''
+        selected_nodes = nuke.selectedNodes()
+        if len(selected_nodes) == 0:
+            selected_nodes = nuke.allNodes()
```

Comment:
All nodes? Shouldn't we show only write nodes?

Reply:
We are creating a write node to simplify a bit, so user do not need to create a write node themselves. Thinks this was the default behaviour of legacy integration (C2 pre framework)